### ISSUE TITLE
openEO oriented JSON types

### ISSUE THREAD

**Issue by soxofaan** [Open-EO/openeo-api#539]
In the JSON world it's common to use a "type" field to indicate what the nature of an object is: e.g. GeoJSON (types "Polygon", "Point", "Feature", ...) or STAC ("Catalog", "Collection).

We're starting to integrate openEO resources outside typical openEO contexts (e.g. remote UDPs #515) and I think it makes sense to start introducing openEO-oriented types for that.

For example, an external UDP should be something like
```json
{
  "type": "openeo-process",
  "id": "evi",
  "process_graph": ...
```

I arbitrarily picked "openeo-process" here.  I'm unaware of any common conventions about that at the moment, so it could equally be "openEO-process", "openEO/process" or whatever

Enforcing this in openEO API 1.x itself is out of the question of course as this would be a breaking change. But it would be good to already settle on some types so that tools can build on that in a future compatible way.

**Comment by m-mohr** [comment]
Do you know any examples outside of GeoJSON-inspired formats? The STAC types are inspired by GeoJSON. I'm wondering whether `type` is somewhat commonly used outside our bubble...

**Comment by soxofaan** [comment]
Indeed, I might be extrapolating from our bubble too much here. But still, openEO has quite some touching points with GeoJSON and STAC, so it still makes sense to adapt some patterns.

Some other comparable solutions I can think of:
- OGC's `conformsTo`
- `$schema` as used in JSON Schema 

but both of these use URIs, which makes it less human friendly than a simple short string
